### PR TITLE
Add RGB-HSV Conversion

### PR DESCRIPTION
**Pull request by algobytewise**
Please include a summary of the change. Please also include relevant motivation and context (if applicable).

- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added comments to hard-to-understand areas of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I have made corresponding changes to the comments
- [x] I have made corresponding changes to the README.md

### CODE REVIEW

**Comment by algobytewise** [comment]
Some comments on the failed Codacy check
* pascal case: It's common to spell RGB & HSV with upper case letters, i.e. non-pascal-case.
* 'out' modifier: This could be avoided by defining two new structs: one for RGB & one for HSV. But the current solution might be preferable in terms of simplicity.
* floating point equality: Just to avoid division by zero

I'll deal with the other errors when I get the time.

**Comment by siriak** [comment]
Hi! Regarding the pascal case, yes, it's common, but it's not standard. See https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions

> The PascalCasing convention, used for all identifiers except parameter names, capitalizes the first character of each word (including acronyms over two letters in length), as shown in the following examples:
> PropertyDescriptor HtmlTag

As for the out modifiers, it's better to just return HSV as a tuple `(double hue, double saturation, double value)`.

I wasn't diving into details with the float comparison issue yet, but I guess you can try comparing initial int value instead of doubles.

**Comment by algobytewise** [comment]
Thanks for looking up the pascal-case guideline. I was able to avoid floating point equality warnings by using `Double.Equals`. Drift shouldn't be a concern since there are no calculations besides Min/Max that are relevant for the comparison.

File: Algorithms/Other/RGBHSVConversion.cs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                red = (byte)Math.Round(255 * (secondLargestComponent + matchValue));
+                green = (byte)Math.Round(255 * matchValue);
+                blue = (byte)Math.Round(255 * (chroma + matchValue));
+            }
+            else
+            {
+                red = (byte)Math.Round(255 * (chroma + matchValue));
+                green = (byte)Math.Round(255 * matchValue);
+                blue = (byte)Math.Round(255 * (secondLargestComponent + matchValue));
+            }
```
Comment (siriak) [inline Algorithms/Other/RGBHSVConversion.cs:136]: Please add a test for that. I think hue > 300 should do the trick

File: README.md [hunk 2]
```diff
@@ -95,6 +95,7 @@ This repository contains algorithms and data structures implemented in C# for ed
 		* [Luhn](./Algorithms/Other/Luhn.cs)
 		* [Mandelbrot](./Algorithms/Other/Mandelbrot.cs)
 		* [Koch Snowflake](./Algorithms/Other/KochSnowflake.cs)
+		* [RGB-HSV Conversion](./Algorithms/Other/RGBHSVConversion.cs)
```
Comment (siriak) [inline README.md:98]: Could you also update https://github.com/TheAlgorithms/C-Sharp/blob/master/DIRECTORY.md ? It's needed for our website to work properly.

**Review by siriak — COMMENTED** [review]
>  I was able to avoid floating point equality warnings by using Double.Equals. Drift shouldn't be a concern since there are no calculations besides Min/Max that are relevant for the comparison.

I agree. == looked cleaner IMO, could you return them back and I'll try to set up Codacy to ignore this issue here?

File: Algorithms/Other/RGBHSVConversion.cs [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                hue = 60 * (2 + (dBlue - dRed) / chroma);
+            }
+            else
+            {
+                hue = 60 * (4 + (dRed - dGreen) / chroma);
+            }
+
+            hue = hue < 0 ? hue + 360 : hue;
+
+            return new Tuple<double, double, double>(hue, saturation, value);
```
[the remark this answers is not in the document]
Comment (algobytewise) [reply Algorithms/Other/RGBHSVConversion.cs]: I get the following error when using the parenthesis-only notation: `Cannot implicitly convert type '(byte red, byte green, byte blue)' to 'System.Tuple<byte, byte, byte>'`. I used the Tuple.Create instead, which is a little bit more readable than the original formulation.

[the remark this answers is not in the document]
Comment (algobytewise) [reply Algorithms/Other/RGBHSVConversion.cs]: It seems the change increased the complexity of the GetRgbBySection-method: see the codacy complexity failure in Algorithms/Other/RGBHSVConversion.cs#L95. This method was already refactored from the HsvToRgb-method. I extracted the conversion to byte into a separate method but that didn't help. I'm not sure if there is a good non-artificial way to further refactor this method.

[the remark this answers is not in the document]
Comment (siriak) [reply Algorithms/Other/RGBHSVConversion.cs]: You should return (double hue, double saturation, double value), not Tuple<double, double, double>. See C# syntax for value tuples. You are using reference tuples but really should value tuples.

**Comment by siriak** [comment]
And could you also return the call to Equals? Sorry for this back and forth :) I can't set up this rule for some reason and don't want to disable it completely because it's valuable. I think having Equals there would be strange, but at the same time, this is good because people will pay extra attention to this place.

**Comment by algobytewise** [comment]
No problem

File: Algorithms/Other/RGBHSVConversion.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (algobytewise) [reply Algorithms/Other/RGBHSVConversion.cs]: Ah I see. I wasn't aware of the difference between regular tuples & value-tuples.

[the remark this answers is not in the document]
Comment (algobytewise) [reply Algorithms/Other/RGBHSVConversion.cs]: It seems the complexity-problem also got solved along the way.

File: Algorithms/Other/RGBHSVConversion.cs [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    public static class RgbHsvConversion
+    {
+        /// <summary>
+        /// Conversion from the HSV-representation to the RGB-representation.
+        /// </summary>
+        /// <param name="hue">Hue of the color.</param>
+        /// <param name="saturation">Saturation of the color.</param>
+        /// <param name="value">Brightness-value of the color.</param>
+        /// <returns>The tuple of RGB-components.</returns>
+        public static ValueTuple<byte, byte, byte> HsvToRgb(
```
Comment (siriak) [inline Algorithms/Other/RGBHSVConversion.cs:25]:
```suggestion
        public static (byte red, byte green, byte blue) HsvToRgb(

```

File: Algorithms/Other/RGBHSVConversion.cs [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        }
+
+        /// <summary>
+        /// Conversion from the RGB-representation to the HSV-representation.
+        /// </summary>
+        /// <param name="red">Red-component of the color.</param>
+        /// <param name="green">Green-component of the color.</param>
+        /// <param name="blue">Blue-component of the color.</param>
+        /// <returns>The tuple of HSV-components.</returns>
+        public static ValueTuple<double, double, double> RgbToHsv(
```
Comment (siriak) [inline Algorithms/Other/RGBHSVConversion.cs:60]:
```suggestion
        public static (double hue, double saturation, double value) RgbToHsv(

```

File: Algorithms/Other/RGBHSVConversion.cs [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+            {
+                hue = 60 * (4 + (dRed - dGreen) / chroma);
+            }
+
+            hue = (hue + 360) % 360;
+
+            return (hue, saturation, value);
+        }
+
+        private static ValueTuple<byte, byte, byte> GetRgbBySection(double hueSection, double chroma, double matchValue, double secondLargestComponent)
```
Comment (siriak) [inline Algorithms/Other/RGBHSVConversion.cs:95]:
```suggestion
        private static (byte red, byte green, byte blue) GetRgbBySection(double hueSection, double chroma, double matchValue, double secondLargestComponent)

```

**Review by siriak — APPROVED** [review]

File: Algorithms/Other/RGBHSVConversion.cs [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+        /// <summary>
+        /// Conversion from the RGB-representation to the HSV-representation.
+        /// </summary>
+        /// <param name="red">Red-component of the color.</param>
+        /// <param name="green">Green-component of the color.</param>
+        /// <param name="blue">Blue-component of the color.</param>
+        /// <returns>The tuple of HSV-components.</returns>
+        public static (double hue, double saturation, double value) RgbToHsv(
+
```
Comment (siriak) [inline Algorithms/Other/RGBHSVConversion.cs:62]:
```suggestion
```

File: Algorithms/Other/RGBHSVConversion.cs [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+                hue = 60 * (4 + (dRed - dGreen) / chroma);
+            }
+
+            hue = (hue + 360) % 360;
+
+            return (hue, saturation, value);
+        }
+
+        private static (byte red, byte green, byte blue) GetRgbBySection(double hueSection, double chroma, double matchValue, double secondLargestComponent)
+
```
Comment (siriak) [inline Algorithms/Other/RGBHSVConversion.cs:98]:
```suggestion
```

File: Algorithms/Other/RGBHSVConversion.cs [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+    {
+        /// <summary>
+        /// Conversion from the HSV-representation to the RGB-representation.
+        /// </summary>
+        /// <param name="hue">Hue of the color.</param>
+        /// <param name="saturation">Saturation of the color.</param>
+        /// <param name="value">Brightness-value of the color.</param>
+        /// <returns>The tuple of RGB-components.</returns>
+        public static (byte red, byte green, byte blue) HsvToRgb(
+
```
Comment (siriak) [inline Algorithms/Other/RGBHSVConversion.cs:26]:
```suggestion
```

**Review by siriak — APPROVED** [review]
Well done!

**Pull request merged**